BGL/include/CGAL/boost/graph/copy_face_graph.h

```diff
@@ -153,8 +154,10 @@ void copy_face_graph_impl(const SourceMesh& sm, TargetMesh& tm,
   }
 
   // update halfedge vertex of all but the vertex halfedge
-  BOOST_FOREACH(tm_vertex_descriptor v, vertices(tm))
+  for(tm_vertex_iterator vit = vertices(tm).first;
+      vit != vertices(tm).second; ++vit)
   {
+    tm_vertex_descriptor v = *vit;
```

Comment:
This was the previous instance of the "`BOOST_FOREACH` problem".
- the loop was not nested in another loop,
- the triangular mesh was indeed modified, but its vector of vertices was not.

Reply:
Could there be something that cause the reallocation of the vector of vertices of the mesh during the loop?

Reply:
- Unless the second issue with `BOOST_FOREACH`: this time MSVC 2017 was impacted (but which update of it)?

Reply:
Is not the loop at https://github.com/janetournois/cgal/blob/9b00be7d7bddee92e6c97f7af089aa0b0a52cc25/Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/repair.h#L283 dangerous?

Reply:
The issue with MSVC is for a user that was using a version of this file that was distributed privately with CGAL 4.9, so way before this fix.

Reply:
The helper function is only using next/opposite to turn around the target vertex

Reply:
All the vertices are created in the loop before so no reallocation after using tm vertex vector.